src/eos-crash-metrics.c

```diff
@@ -137,6 +180,118 @@ normalize_path (char *path)
     path[i] = path[i] == '!' ? '/' : path[i];
 }
 
+static inline void
+sanitize_name (gchar *name)
+{
+  gchar *tmp = strchr (name, '/');
+
+  if (tmp != NULL)
+    *tmp = '\0';
+}
+
+static const char *
+get_associated_runtime (FlatpakInstalledRef *ref)
+{
+  g_autoptr(GBytes) metadata = NULL;
+  gchar *runtime = NULL;
+  g_autoptr(GKeyFile) key_file = NULL;
+  const gchar *metadata_data;
+  gsize metadata_len;
+  g_autoptr(GError) local_error = NULL;
+
+  metadata = flatpak_installed_ref_load_metadata (FLATPAK_INSTALLED_REF (ref), NULL, &local_error);
+
+  if (metadata == NULL)
+    {
+      g_warning ("%s", local_error->message);
+      return NULL;
+    }
+
+  metadata_data = g_bytes_get_data (metadata, &metadata_len);
+  key_file = g_key_file_new ();
+
+  if (!g_key_file_load_from_data (key_file, metadata_data, metadata_len, G_KEY_FILE_NONE, &local_error))
+    {
+      if (!g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        g_warning ("%s", local_error->message);
+
+      return NULL;
+    }
+
+  if (!(runtime = g_key_file_get_string (key_file, "Application", "runtime", NULL)))
+    return NULL;
+
+  sanitize_name (runtime);
+  return runtime;
+}
+
+static FlatpakInfo *
+get_flatpak_info (const char *path,
+                  GError **error)
+{
+  FlatpakInfo *info;
+  g_autoptr(GPtrArray) xrefs = NULL;
+  g_autoptr(GPtrArray) runtimes = NULL;
+  guint i;
+  g_autofree char *executable_name = NULL;
+  g_autoptr(FlatpakInstallation) installation = flatpak_installation_new_system (NULL, error);
+
+  if (installation == NULL)
+    return NULL;
+
+  executable_name = g_path_get_basename (path);
+
+  info = flatpak_info_new();
+
+  /* get apps and runtimes */
+  xrefs = flatpak_installation_list_installed_refs (installation,
+                                                    NULL, error);
+  if (xrefs == NULL)
+    return NULL;
```

Comment:
`info` is leaked here, consider making it `autoptr`

Reply:
Doesn't the caller takes the ownership? And that is an "g_autoptr". Fwiw, Valgrind seems to be happy, too.